AppCenterDataStorage/AppCenterDataStorage/MSDataStore.m

```diff
@@ -272,40 +367,41 @@ - (void)deleteDocumentWithPartition:(NSString *)partition
                          documentId:(NSString *)documentId
                        writeOptions:(MSWriteOptions *)__unused writeOptions
                   completionHandler:(MSDataSourceErrorCompletionHandler)completionHandler {
+
   // TODO consume writeOptions
   [MSTokenExchange
       performDbTokenAsyncOperationWithHttpClient:(MSStorageIngestion *)self.ingestion
                                        partition:partition
                                completionHandler:^(MSTokensResponse *_Nonnull tokenResponses, NSError *_Nonnull tokenExchangeError) {
-
                                  // If error getting token.
                                  if (tokenExchangeError || [tokenResponses.tokens count] == 0) {
-                                   NSNumber *httpStatusCode = [tokenExchangeError userInfo][kMSCosmosDbHttpCodeKey];
-                                     MSLogError([MSDataStore logTag], @"Can't get CosmosDb token. Error: %@;  HTTP status code: %d; Partition: %@", tokenExchangeError.localizedDescription,
-                                              [httpStatusCode intValue], partition);
+                                   NSInteger httpStatusCode = [MSDataSourceError errorCodeWithError:tokenExchangeError];
+                                   MSLogError([MSDataStore logTag],
+                                              @"Can't get CosmosDb token. Error: %@;  HTTP status code: %ld; Partition: %@",
+                                              tokenExchangeError.localizedDescription, (long)httpStatusCode, partition);
```

Comment:
What is the reason casting `httpStatusCode` to `long` instead of `int`?

Reply:
Per https://stackoverflow.com/a/16076218/873234, it's recommended to cast `NSInteger` to `long` to be formatted correctly on 32-bit/64-bit system.
> Since NSInteger is 32-bit or 64-bit, depending on the platform, the compiler recommends to add a cast to long generally.